### PR TITLE
Add support for custom apt-daily and apt-daily-upgrade timers

### PR DESCRIPTION
**Pull request by pgassmann**
Restore of #86 by @mabed-fr

with additional fixes and improvements based on the feedback on the original PR.

### CODE REVIEW

**Comment by pgassmann** [comment]
@christianhueserhzdr You reviewed the orignal PR from which I incorporated your feedback. Can you check the changes I did?

File: tasks/systemd_timers.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+---
+
+- name: 'Test apt-daily timer expression'
+  ansible.builtin.command:
+    cmd: 'systemd-analyze calendar "{{ unattended_apt_daily_oncalendar }}"'
+  register: '_apt_daily'
+  changed_when: '_apt_daily.rc != 0'
+  check_mode: 'no'
```
Comment (Normo) [inline tasks/systemd_timers.yml:8]:
```suggestion
  check_mode: false
```

File: tasks/systemd_timers.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+- name: 'Test apt-daily-upgrade timer expression'
+  ansible.builtin.command:
+    cmd: 'systemd-analyze calendar "{{ unattended_apt_daily_upgrade_oncalendar }}"'
+  register: '_apt_daily_upgrade'
+  changed_when: '_apt_daily_upgrade.rc != 0'
+  check_mode: 'no'
```
Comment (Normo) [inline tasks/systemd_timers.yml:15]:
```suggestion
  check_mode: false
```

File: tasks/systemd_timers.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    - 'Restart apt-daily timer'
+
+- name: 'Deploy apt-daily-upgrade timer'
+  ansible.builtin.template:
+    src: 'apt_daily_upgrade_override.conf.j2'
+    dest: '/etc/systemd/system/apt-daily-upgrade.timer.d/schedule_override.conf'
+    mode: '0644'
+  when:
+    - '_apt_daily_upgrade.rc == 0'
+    - '_apt_daily_upgrade_timer_d' # skip if run for the first  time in checkmode
```
Comment (Normo) [inline tasks/systemd_timers.yml:49]:
```suggestion
    - '_apt_daily_upgrade_timer_d'  # skip if run for the first time in check mode
```

File: tasks/systemd_timers.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  register: '_apt_daily_upgrade_timer_d'
+
+- name: 'Deploy apt-daily timer'
+  ansible.builtin.template:
+    src: 'apt_daily_override.conf.j2'
+    dest: '/etc/systemd/system/apt-daily.timer.d/schedule_override.conf'
+    mode: '0644'
+  when:
+    - '_apt_daily.rc == 0'
+    - '_apt_daily_timer_d' # skip if run for the first time in checkmode
```
Comment (Normo) [inline tasks/systemd_timers.yml:38]:
```suggestion
    - '_apt_daily_timer_d'  # skip if run for the first time in check mode
```

**Review by Normo — CHANGES_REQUESTED** [review]
Thank you very much @pgassmann! I made a few suggestions that should be implemented before we can merge it.

File: README.md [hunk 5]
```diff
@@ -148,6 +163,41 @@ Additionally, unattended-upgrades support two macros (variables), derived from `
...
+unattended_apt_daily_oncalendar: "*-*-* 6,18:00" # (default)
+unattended_apt_daily_randomizeddelaysec: "12h" # (default)
+# apt-daily-upgrade timer
+unattended_apt_daily_upgrade_oncalendar: "*-*-* 6:00" # (default)
+unattended_apt_daily_upgrade_randomizeddelaysec: "60m" # (default)
+```
+
+### Override Example
+
+* Download is between 12:20am - 12:40am
```
Comment (pgassmann) [inline README.md:187]: this seems wrong to me. According to the docs, the RandomizedDelaySec should only be added.

File: README.md [hunk 6]
```diff
@@ -148,6 +163,41 @@ Additionally, unattended-upgrades support two macros (variables), derived from `
 
 Using `${distro_codename}` should be preferred over using `stable` or `oldstable` as a selected, as once `stable` moves to `oldstable`, no security updates will be installed at all, or worse, package from a newer distro release will be installed by accident. The same goes for upgrading your installation from `oldstable` to `stable`, if you forget to change this in your origin patterns, you may not receive the security updates for your newer distro release. With `${distro_codename}`, both cases can never happen.
 
+## Systemd timers
+
+Documentation for systemd/Timers: <https://wiki.archlinux.org/title/systemd/Timers>
+
+### Debian Default Configuration
+
+* Download daily between at random times during the entire day.
```
Comment (Normo) [inline README.md:172]:
```suggestion
* Download daily at random times during the entire day.
```

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -49,7 +49,7 @@ jobs:
         uses: "actions/checkout@v3"
 
       - name: "Enable lingering required for podman and systemd in GH Actions."
-        run: |  # Compare with https://github.com/eriksjolund/user-systemd-service-actions-workflow/blob/efe872924fd2dd35bb482544126ce751303e14c2/README.md
+        run: | # Compare with https://github.com/eriksjolund/user-systemd-service-actions-workflow/blob/efe872924fd2dd35bb482544126ce751303e14c2/README.md
```
Comment (Normo) [inline .github/workflows/ci.yml:52]:
```suggestion
        run: |  # Compare with https://github.com/eriksjolund/user-systemd-service-actions-workflow/blob/efe872924fd2dd35bb482544126ce751303e14c2/README.md
```

File: tasks/systemd_timers.yml [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    cmd: 'systemd-analyze calendar "{{ unattended_apt_daily_upgrade_oncalendar }}"'
+  register: '_apt_daily_upgrade'
+  changed_when: '_apt_daily_upgrade.rc != 0'
+  check_mode: false
+
+- name: 'Ensure directory apt-daily.timer.d exists'
+  ansible.builtin.file:
+    path: '/etc/systemd/system/apt-daily.timer.d'
+    mode: '0755'
+    state: 'directory'
```
Comment (Normo) [inline tasks/systemd_timers.yml:21]:
```suggestion
    state: 'directory'
    owner: 'root'
    group: 'root'
```

File: tasks/systemd_timers.yml [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+  register: '_apt_daily_timer_d'
+
+- name: 'Ensure directory apt-daily-upgrade.timer.d exists'
+  ansible.builtin.file:
+    path: '/etc/systemd/system/apt-daily-upgrade.timer.d'
+    mode: '0755'
+    state: 'directory'
```
Comment (Normo) [inline tasks/systemd_timers.yml:28]:
```suggestion
    state: 'directory'
    owner: 'root'
    group: 'root'
```

File: tasks/systemd_timers.yml [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+  register: '_apt_daily_upgrade_timer_d'
+
+- name: 'Deploy apt-daily timer'
+  ansible.builtin.template:
+    src: 'apt_daily_override.conf.j2'
+    dest: '/etc/systemd/system/apt-daily.timer.d/schedule_override.conf'
+    mode: '0644'
```
Comment (Normo) [inline tasks/systemd_timers.yml:35]:
```suggestion
    mode: '0644'
    owner: 'root'
    group: 'root'
```

File: tasks/systemd_timers.yml [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+    - '_apt_daily.rc == 0'
+    - '_apt_daily_timer_d' # skip if run for the first time in check mode
+  notify:
+    - 'Restart apt-daily timer'
+
+- name: 'Deploy apt-daily-upgrade timer'
+  ansible.builtin.template:
+    src: 'apt_daily_upgrade_override.conf.j2'
+    dest: '/etc/systemd/system/apt-daily-upgrade.timer.d/schedule_override.conf'
+    mode: '0644'
```
Comment (Normo) [inline tasks/systemd_timers.yml:46]:
```suggestion
    mode: '0644'
    owner: 'root'
    group: 'root'
```

**Review by Normo — APPROVED** [review]
Thanks @pgassmann

**Pull request merged**